File: aat-lib/src/main/java/ch/bailu/aat_lib/description/DateDescription.java
package ch.bailu.aat_lib.description;


import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.resources.Res;

public class DateDescription extends LongDescription {
    @Override
    public String getLabel() {
        return Res.str().d_startdate();
    }

    @Override
    public String getValue()   {
        return FF.f().LOCAL_DATE_TIME.format(getCache());
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        setCache(info.getStartTime());
    }

}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/DateDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.information.GpxInformation
import ch.bailu.aat_lib.resources.Res

open class DateDescription : LongDescription() {
    override fun getLabel(): String {
        return Res.str().d_startdate()
    }

    override fun getValue(): String {
        return FormatDisplay.f().localDateTime.format(cache)
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        setCache(info.getStartTime())
    }
}
